sale_order_safe_commitment_date/models/sale_order.py

```diff
@@ -0,0 +1,30 @@
+# Copyright 2025 Moduon Team S.L.
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl-3.0)
+from odoo import api, fields, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    is_commitment_date_unsafe = fields.Boolean(
+        compute="_compute_is_commitment_date_unsafe",
+    )
+
+    @api.depends("commitment_date", "expected_date", "state")
+    def _compute_is_commitment_date_unsafe(self):
+        """A commitment date is considered unsafe if it is before the expected date as
+        the products won't be delivered on time."""
+        self.is_commitment_date_unsafe = False
+        for order in self.filtered(
+            lambda x: x.commitment_date
+            and x.expected_date
+            and x.state in {"draft", "sent"}
+        ):
+            order.is_commitment_date_unsafe = (
```

Comment:
Putting the condition in `filtered` will allow to do just one `__set__` on the field for `True` values.

Reply:
Done, thanks :)